File: src/TOMI/models/dto/Badge.dto.ts
export interface BadgeResponseDto {
  badgeId: number;
  name: string;
  description: string;
  imageUrl: string;
  criteria: string;
}

export interface BadgeCreateDto {
  name: string;
  description: string;
  imageUrl: string;
  criteria: string;
}

export interface BadgeUpdateDto {
  name?: string;
  description?: string;
  imageUrl?: string;
  criteria?: string;
}

File: src/Backend/Core/DTO/BadgeDTO.py
from typing import Optional
from pydantic import BaseModel, Field

# DTO for creating a new badge
class BadgeCreateDTO(BaseModel):
    achievement: str = Field(..., min_length=1, max_length=255)
    name: str = Field(..., min_length=1, max_length=255)
    description: str = Field(..., min_length=1)
    
    
    
# DTO for updating a badge
class BadgeUpdateDTO(BaseModel):
    achievement: Optional[str] = Field(None, min_length=1, max_length=255)
    name: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = Field(None, min_length=1)
    

# DTO for responding with badge data
class BadgeResponseDTO(BaseModel):
    badgeId: int
    achievement: str
    name: str
    description: str
    

